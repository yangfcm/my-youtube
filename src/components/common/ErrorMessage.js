import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const renderError = (error) => {
  return (
    <div className="text-danger text-center my-3 py-3">
      <FontAwesomeIcon icon="exclamation-triangle" className="mx-2" />
      {error.displayMessage}
    </div>
  );
};

const ErrorMessage = ({ error }) => {
  // console.log(process.env);
  if (process.env.NODE_ENV === "development") {
    console.log(error);
  }
  return renderError(error);
};

export default ErrorMessage;
