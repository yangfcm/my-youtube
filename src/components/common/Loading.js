import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

const Loading = () => {
  return (
    <div className="text-center my-3 py-3">
      <FontAwesomeIcon icon="spinner" className="mx-2" />
      Loading...
    </div>
  );
};

export default Loading;
