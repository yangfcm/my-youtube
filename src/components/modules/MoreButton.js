import React from "react";

const MoreButton = ({ onClickMore, children }) => {
  return (
    <div className="text-center">
      <button
        className="btn btn-light"
        style={{ width: "100%" }}
        onClick={onClickMore}
      >
        <span className="text-dark">{children ? children : "More..."}</span>
      </button>
    </div>
  );
};

export default MoreButton;
