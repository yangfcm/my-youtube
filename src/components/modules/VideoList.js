import React from "react";
import VideoListItem from "../modules/VideoListItem";

const renderList = (videoList, playlistId) => {
  return videoList.map(video => {
    return (
      <VideoListItem key={video.id} video={video} playlistId={playlistId} />
    );
  });
};

const VideoList = ({ videoList, playlistId }) => {
  return <div>{renderList(videoList, playlistId)}</div>;
};

export default VideoList;
