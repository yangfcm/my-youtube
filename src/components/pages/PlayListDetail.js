import React from "react";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import VideoList from "../modules/VideoList";
import MoreButton from "../modules/MoreButton";
import Loading from "../common/Loading";
import ErrorMessage from "../common/ErrorMessage";

import { fetchPlaylistDetail } from "../../actions/playlist";
import { clearError } from "../../actions/error";

export class PlayListDetail extends React.Component {
  state = {
    playlistDetail: null,
    error: null,
  };

  componentDidMount = async () => {
    await this.props.fetchPlaylistDetail(this.props.match.params.id);
    if (this.props.error) {
      this.setState({
        error: this.props.error,
      });
      return;
    }
    if (this.props.playlistDetail) {
      this.setState({
        playlistDetail: {
          pageInfo: this.props.playlistDetail.pageInfo,
          items: this.props.playlistDetail.items,
          nextPageToken: this.props.playlistDetail.nextPageToken,
        },
      });
    }
  };

  componentWillUnmount = () => {
    this.props.clearError();
    this.setState({
      playlistDetail: null,
    });
  };

  fetchNextPagePlaylist = async () => {
    const { nextPageToken } = this.state.playlistDetail;
    await this.props.fetchPlaylistDetail(
      this.props.match.params.id,
      nextPageToken
    );
    if (this.props.error) {
      this.setState({
        error: this.props.error,
      });
      return;
    }
    this.setState((state, props) => {
      return {
        playlistDetail: {
          pageInfo: props.playlistDetail.pageInfo,
          items: state.playlistDetail.items.concat(props.playlistDetail.items),
          nextPageToken: props.playlistDetail.nextPageToken,
        },
      };
    });
  };

  render() {
    return (
      <React.Fragment>
        {!this.state.error && !this.state.playlistDetail && <Loading />}

        {this.state.error && <ErrorMessage error={this.props.error} />}

        {this.state.playlistDetail && (
          <div className="d-flex justify-content-center">
            <div className="col-lg-9">
              <div className="mb-3">
                <h5 className="font-weight-bold">Playlist Videos</h5>
              </div>
              <VideoList
                videoList={this.state.playlistDetail.items}
                playlistId={this.props.match.params.id}
              />

              {this.state.playlistDetail.nextPageToken && (
                <div style={{ width: "50%", margin: "0 auto" }}>
                  <MoreButton onClickMore={this.fetchNextPagePlaylist} />
                </div>
              )}
            </div>
          </div>
        )}
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    playlistDetail: state.playlist.playlistDetail,
    error: state.error,
  };
};

export default connect(mapStateToProps, { fetchPlaylistDetail, clearError })(
  withRouter(PlayListDetail)
);
