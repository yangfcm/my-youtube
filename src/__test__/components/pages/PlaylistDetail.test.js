import React from "react";
import { shallow } from "enzyme";
import { PlayListDetail } from "components/pages/PlayListDetail";
import VideoList from "components/modules/VideoList";
import MoreButton from "components/modules/MoreButton";
import Loading from "components/common/Loading";
import ErrorMessage from "components/common/ErrorMessage";
import {
  playlistDetailResponse,
  playlistErrorResponse,
  playlistId,
} from "__test__/fixtures/playlist";

describe("Test PlaylistDetail component", () => {
  let wrapper;
  const mockFetchPlaylistDetail = jest.fn();
  const mockClearError = jest.fn();

  afterEach(() => {
    wrapper.unmount();
  });

  describe("When component is initially rendered", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayListDetail
          fetchPlaylistDetail={mockFetchPlaylistDetail}
          clearError={mockClearError}
          match={{ params: { id: playlistId } }}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render Loading component", () => {
      expect(wrapper.find(Loading).exists()).toBe(true);
    });

    it("should fetch playlist detail", () => {
      expect(mockFetchPlaylistDetail).toHaveBeenCalledWith(playlistId);
    });
  });

  describe("When error occurs", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayListDetail
          fetchPlaylistDetail={mockFetchPlaylistDetail}
          clearError={mockClearError}
          match={{ params: { id: playlistId } }}
          error={playlistErrorResponse}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render ErrorMessage component", () => {
      expect(wrapper.find(ErrorMessage).exists()).toBe(true);
    });
  });

  describe("When videos in the playlist is fetched", () => {
    beforeEach(() => {
      wrapper = shallow(
        <PlayListDetail
          fetchPlaylistDetail={mockFetchPlaylistDetail}
          clearError={mockClearError}
          match={{ params: { id: playlistId } }}
          error={null}
          playlistDetail={playlistDetailResponse}
        />
      );
    });

    it("should render component", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render VideoList component and pass videos and playlist id to it", () => {
      const videoList = wrapper.find(VideoList);
      expect(videoList.exists()).toBe(true);
      expect(videoList.prop("videoList")).toEqual(playlistDetailResponse.items);
      expect(videoList.prop("playlistId")).toBe(playlistId);
    });

    it("should render more button and respond to click", () => {
      const moreButton = wrapper.find(MoreButton);
      expect(moreButton.exists()).toBe(true);
      moreButton.prop("onClickMore")();
      expect(mockFetchPlaylistDetail).toHaveBeenCalledWith(
        playlistId,
        playlistDetailResponse.nextPageToken
      );
    });
  });
});
