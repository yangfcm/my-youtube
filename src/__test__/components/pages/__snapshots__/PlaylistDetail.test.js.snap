// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlaylistDetail component When component is initially rendered should render component 1`] = `
<Fragment>
  <Loading />
</Fragment>
`;

exports[`Test PlaylistDetail component When error occurs should render component 1`] = `
<Fragment>
  <ErrorMessage
    error={
      Object {
        "response": Object {
          "data": Object {
            "error": Object {
              "code": 404,
              "errors": Array [
                Object {
                  "domain": "youtube.playlist",
                  "location": "channelId",
                  "locationType": "parameter",
                  "message": "Channel not found.",
                  "reason": "channelNotFound",
                },
              ],
              "message": "Channel not found.",
            },
          },
        },
      }
    }
  />
</Fragment>
`;

exports[`Test PlaylistDetail component When videos in the playlist is fetched should render component 1`] = `
<Fragment>
  <div
    className="d-flex justify-content-center"
  >
    <div
      className="col-lg-9"
    >
      <div
        className="mb-3"
      >
        <h5
          className="font-weight-bold"
        >
          Playlist Videos
        </h5>
      </div>
      <VideoList
        playlistId="PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG"
        videoList={
          Array [
            Object {
              "contentDetails": Object {
                "videoId": "SqrbIlUwR0U",
                "videoPublishedAt": "2018-09-18T20:17:43Z",
              },
              "etag": "y9NNPhfiMScskVn1q-6fsAqr1OI",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy41NkI0NEY2RDEwNTU3Q0M2",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "In this crash course we will get started with the Go programming language. There is no prior knowledge of Go required. We will setup a workspace and look at fundamentals like variables, slices, conditionals, maps, structs and more

Workspace - 1:35
Hello World - 7:37
Variable & Types - 11:18
Packages - 21:43
Functions - 26:25
Arrays / Slices - 29:35
Conditionals - 35:15
Loops - 41:30
Maps - 47:27
Range - 0:53:25
Pointers - 1:01:23
Closures - 1:06:48
Structs - 1:10:54
Interfaces - 1:26:53
Web - 1:33:29

Sponsor: DevMountain Bootcamp
https://goo.gl/6q0dEa

Code:
https://github.com/bradtraversy/go_crash_course (I plan on tidying the code up later)

Doc:
https://golang.org/doc/code.html

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Website & Udemy Courses
http://www.traversymedia.com

Follow Traversy Media:
https://www.facebook.com/traversymedia
https://www.twitter.com/traversymedia
https://www.instagram.com/traversymedia",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 0,
                "publishedAt": "2018-09-21T12:15:42Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "SqrbIlUwR0U",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/SqrbIlUwR0U/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/SqrbIlUwR0U/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/SqrbIlUwR0U/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/SqrbIlUwR0U/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/SqrbIlUwR0U/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Go / Golang Crash Course",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "PoRJizFvM7s",
                "videoPublishedAt": "2018-06-22T13:13:12Z",
              },
              "etag": "hbbvSOez96ehPNfM_N9X8PoqK7Y",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy4yODlGNEE0NkRGMEEzMEQy",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "In this crash course we will look at asynchronous JavaScript and cover callbacks, promises including promise.all as well as the async / await syntax.

FULL JS COURSE (Just 10 Bucks!!!!):
https://www.udemy.com/modern-javascript-from-the-beginning/?couponCode=YOUTUBE10

CODE:
https://plnkr.co/4Pv2HhiWV4kiPHY9VrUw

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Follow Traversy Media:
http://www.facebook.com/traversymedia
http://www.twitter.com/traversymedia
http://www.instagram.com/traversymedia",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 1,
                "publishedAt": "2018-09-21T12:25:16Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "PoRJizFvM7s",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/PoRJizFvM7s/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/PoRJizFvM7s/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/PoRJizFvM7s/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/PoRJizFvM7s/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/PoRJizFvM7s/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Async JS Crash Course - Callbacks, Promises, Async Await",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "bjVUqvcCnxM",
                "videoPublishedAt": "2018-09-29T14:28:45Z",
              },
              "etag": "_IG8Ximangw3SLpnFPLe8EZWQxw",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy4wMTcyMDhGQUE4NTIzM0Y5",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "In this video I will show you how to deploy a website or frontend application to Netlify which is a fantastic service for hosting with free SSL, server-side functions and even built in form functionality. This video is NOT sponsored by Netlify in any way

Sponsor: Freelancer Bundle (Use \\"brad25\\" for 25% off)
https://studywebdevelopment.com/freelancing.html

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Website & Udemy Courses
http://www.traversymedia.com

Follow Traversy Media:
https://www.facebook.com/traversymedia
https://www.twitter.com/traversymedia
https://www.instagram.com/traversymedia",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 2,
                "publishedAt": "2018-10-02T14:09:53Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "bjVUqvcCnxM",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/bjVUqvcCnxM/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/bjVUqvcCnxM/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/bjVUqvcCnxM/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/bjVUqvcCnxM/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/bjVUqvcCnxM/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Deploy Websites In Seconds With Netlify",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "SKXkC4SqtRk",
                "videoPublishedAt": "2018-09-11T16:30:26Z",
              },
              "etag": "rvyHu2dXrV0Mvlr6yIMIH1C16Ik",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy41MjE1MkI0OTQ2QzJGNzNG",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "In this video we will use the gh-pages npm module to easily deploy any frontend app or website to Github pages. We will do a static site and a React app. I will also show you how to connect a domain name.

Sponsor: DevMountain Bootcamp
https://goo.gl/6q0dEa

Github A Record IPs:
185.199.108.153
185.199.109.153
185.199.110.153
185.199.111.153

Get Domains Cheap: Please use affiliate link :)
https://namecheap.pxf.io/c/1299552/386170/5618

Git Crash Course:
https://www.youtube.com/watch?v=SWYqp7iY_Tc

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Website & Udemy Courses
http://www.traversymedia.com

Follow Traversy Media:
https://www.facebook.com/traversymedia
https://www.twitter.com/traversymedia
https://www.instagram.com/traversymedia",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 3,
                "publishedAt": "2018-10-02T14:10:39Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "SKXkC4SqtRk",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/SKXkC4SqtRk/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/SKXkC4SqtRk/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/SKXkC4SqtRk/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/SKXkC4SqtRk/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/SKXkC4SqtRk/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "GitHub Pages Deploy & Domain",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "ubCNZRNjhyo",
                "videoPublishedAt": "2017-06-08T15:00:04Z",
              },
              "etag": "JOKYLGidCqLQkK_YYwhoTmmFNfE",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy4wOTA3OTZBNzVEMTUzOTMy",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "Let's dive right into the basics of AWS so that you get a quick overview over what AWS actually is and what you can do with it.
Join the full AWS Serverless course: https://acad.link/aws-serverless
Join our Academind Community on Discord for free: https://discord.gg/gxvEWGU

Dive into the complete AWS Basics series: https://academind.com/learn/aws/the-basics/

Check out all our other courses: https://academind.com/learn/our-courses

Want to become a frontend developer? Dive into one of these courses!
Ionic - The Practical Guide: https://acad.link/ionic
Angular - The Complete Guide: https://acad.link/angular
Vue.js - The Complete Guide: https://acad.link/vuejs

----------

Example Project (don't forget to unzip + rezip!): https://github.com/academind/aws-first-node-app

Some useful AWS Articles:
- Getting Started: https://aws.amazon.com/getting-started
- Infrastructure: https://aws.amazon.com/about-aws/global-infrastructure
- Pricing: https://aws.amazon.com/pricing
- More on EC2 (since I show it in the video): https://aws.amazon.com/ec2
- More on Elastic Beanstalk: https://aws.amazon.com/elasticbeanstalk

----------

â€¢ Go to https://www.academind.com and subscribe to our newsletter to stay updated and to get exclusive content & discounts
â€¢ Follow @maxedapps and @academind_real on Twitter
â€¢ Join our Facebook community on https://www.facebook.com/academindchannel/

See you in the videos!

----------

Academind is your source for online education in the areas of web development, frontend web development, backend web development, programming, coding and data science! No matter if you are looking for a tutorial, a course, a crash course, an introduction, an online tutorial or any related video, we try our best to offer you the content you are looking for. Our topics include Angular, React, Vue, Html, CSS, JavaScript, TypeScript, Redux, Nuxt.js, RxJs, Bootstrap, Laravel, Node.js, Progressive Web Apps (PWA), Ionic, React Native, Regular Expressions (RegEx), Stencil, Power BI, Amazon Web Services (AWS), Firebase or other topics, make sure to have a look at this channel or at academind.com to find the learning resource of your choice!

xxx

Let's dive right into the basics of AWS so that you get a quick overview over what AWS actually is, what you can do with it and how it works!

Want to learn AWS Serverless apps? Dive into my complete introduction for only $13: https://www.udemy.com/aws-serverless-a-complete-introduction/?couponCode=YOUTUBE_1

Example Project (don't forget to unzip + rezip!): https://github.com/academind/aws-first-node-app

Some useful AWS Articles:
- Getting Started: https://aws.amazon.com/getting-started
- Infrastructure: https://aws.amazon.com/about-aws/global-infrastructure
- Pricing: https://aws.amazon.com/pricing
- More on EC2 (since I show it in the video): https://aws.amazon.com/ec2
- More on Elastic Beanstalk: https://aws.amazon.com/elasticbeanstalk

Want to become a frontend developer? Consider diving into some of my courses on Udemy:
Angular vs React vs Vue - Quickstart and Comparison for $12: https://www.udemy.com/angular-reactjs-vuejs-quickstart-comparison/?couponCode=YOUTUBE_PROMO
Ionic 2 - The Practical Guide for only $15: https://www.udemy.com/ionic-2-the-practical-guide-to-building-ios-android-apps/?couponCode=GETINTOIT
Angular - The Complete Guide for only $15, too: https://www.udemy.com/the-complete-guide-to-angular-2/?couponCode=YOUTUBE_2
Vue.js 2 - The Complete Guide for only $15: https://www.udemy.com/vuejs-2-the-complete-guide/?couponCode=YOUTUBE_VUE

Want to get some 1-on-1 coaching with experienced developers? Have a look at Savvy: https://www.savvy.is/?ref=9fee2b (Discount Code 9fee2b)

You can follow me on Twitter (@maxedapps), Facebook (https://www.facebook.com/academindchannel/) or visit our Website (https://www.academind.com).

See you in the videos!",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 4,
                "publishedAt": "2018-12-04T10:17:22Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "ubCNZRNjhyo",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/ubCNZRNjhyo/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/ubCNZRNjhyo/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/ubCNZRNjhyo/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/ubCNZRNjhyo/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/ubCNZRNjhyo/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Getting Started with AWS | Amazon Web Services BASICS",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "UnTQVlqmDQ0",
                "videoPublishedAt": "2018-12-20T14:19:56Z",
              },
              "etag": "IIXKZ9HPoWQoJxlni5uyNY9v_tM",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy4xMkVGQjNCMUM1N0RFNEUx",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "This is my yearly step by step guide to becoming a web developer in 2019. We will look at nearly all aspects of web technology including the necessities as well as some of the new trends for 2019.

New 2020 Guide - https://www.youtube.com/watch?v=0pThnRneDjw&t=2s

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Website & Udemy Courses
http://www.traversymedia.com

Follow Traversy Media:
https://www.facebook.com/traversymedia
https://www.twitter.com/traversymedia
https://www.instagram.com/traversymedia

Shared & VPS Hosting:
https://inmotion-hosting.evyy.net/c/396530/260033/4222

Cheap Domain Names:
https://namecheap.pxf.io/c/1299552/386170/5618",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 5,
                "publishedAt": "2018-12-20T20:03:30Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "UnTQVlqmDQ0",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/UnTQVlqmDQ0/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/UnTQVlqmDQ0/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/UnTQVlqmDQ0/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/UnTQVlqmDQ0/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/UnTQVlqmDQ0/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Web Development In 2019 - A Practical Guide",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "PK30-hguZ_Y",
                "videoPublishedAt": "2018-06-15T11:30:00Z",
              },
              "etag": "in_BF70Y7EmJwulKPlR-GfJz9Os",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy41MzJCQjBCNDIyRkJDN0VD",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "NEW UPDATED YOAST TUTORIAL HERE (OCTOBER 2019) https://www.youtube.com/watch?v=ihymaUI1ErY&t=192s
LEARN HOW TO MAKE A WORDPRESS WEBSITE FOR 2019 HERE -- https://www.youtube.com/watch?v=rCcd6IcmA64
NEW Coupon Code For Yoast Courses: DW6762-15% ( enter coupon code at checkout)
Join The 1st Wordpress Social Platform! Register NOW: https://www.wpfriends.com
Learn How To Setup Yoast With Your Wordpress Website. The Yoast Plugin Will Help Increase Your Website Presence and Visibility!

Important Links Below:

Download Yoast Here(Or Purchase Pro): https://www.darrelwilson.com/yoast
FREE YOAST Course: https://yoast.com/academy/courses/#course_free-seo-course-seo-for-beginners
Premium SEO Course Here: https://yoast.com/academy/courses/
Google Web Master Tools Link: https://www.google.com/webmasters


Content Writers Or SEO Help

iWriter - (Very Low Cost) https://www.darrelwilson.com/iwriter
Scripted ( Higher Perfomance Blogs) https://www.darrelwilson.com/scripted
Outsource Your SEO: http://www.darrelwilson.com/freelancer

Also if you liked the layout, i used the elementor page builder and crocoblocks to create those layouts!

Visit Crocoblock: https://www.darrelwilson.com/crocoblock

Learn how to use the amazing yoast plugin! The yoast plugin helps optimize your wordpress website for search engines like google and bing.
 
You can change the way your website appears in the search engines. It can really increase the organic growth of your website as well. In this yoast seo tutorial, i will cover all the things you need to know about the yoast SEO plugin!

The yoast plugin also has features for people to share and identify your website with image sharing, another great way to get more exposure! 

For more tutorials like this, visit my website at http://www.darrelwilson.com",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 6,
                "publishedAt": "2019-03-19T13:33:19Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "PK30-hguZ_Y",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/PK30-hguZ_Y/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/PK30-hguZ_Y/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/PK30-hguZ_Y/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/PK30-hguZ_Y/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/PK30-hguZ_Y/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "Yoast Seo Tutorial 2020 - How To Setup Yoast SEO Plugin - Wordpress SEO By Yoast",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
            Object {
              "contentDetails": Object {
                "videoId": "b6Oe2puTdMQ",
                "videoPublishedAt": "2019-04-10T11:50:19Z",
              },
              "etag": "o9h5Vi8rI2i5ld4o41AySHRgfio",
              "id": "UExsN195bG80Y0hBU01CSzFWNVdFZ3dvZzVJb01ORUxVRy5DQUNERDQ2NkIzRUQxNTY1",
              "kind": "youtube#playlistItem",
              "snippet": Object {
                "channelId": "UC4ZrkawfeduKHUrm_K1RUag",
                "channelTitle": "Fan Yang",
                "description": "In this project we will build a full stack file upload with React and Express. We will construct the UI with a Bootstrap elements and make them function

Sponsor: DevMountain Bootcamp
https://goo.gl/6q0dEa

Code:
https://github.com/bradtraversy/react_file_uploader

ðŸ’– Become a Patron: Show support & get perks!
http://www.patreon.com/traversymedia

Website & Udemy Courses
http://www.traversymedia.com

Follow Traversy Media:
https://www.facebook.com/traversymedia
https://www.twitter.com/traversymedia
https://www.instagram.com/traversymedia",
                "playlistId": "PLl7_ylo4cHASMBK1V5WEgwog5IoMNELUG",
                "position": 7,
                "publishedAt": "2019-04-10T21:51:22Z",
                "resourceId": Object {
                  "kind": "youtube#video",
                  "videoId": "b6Oe2puTdMQ",
                },
                "thumbnails": Object {
                  "default": Object {
                    "height": 90,
                    "url": "https://i.ytimg.com/vi/b6Oe2puTdMQ/default.jpg",
                    "width": 120,
                  },
                  "high": Object {
                    "height": 360,
                    "url": "https://i.ytimg.com/vi/b6Oe2puTdMQ/hqdefault.jpg",
                    "width": 480,
                  },
                  "maxres": Object {
                    "height": 720,
                    "url": "https://i.ytimg.com/vi/b6Oe2puTdMQ/maxresdefault.jpg",
                    "width": 1280,
                  },
                  "medium": Object {
                    "height": 180,
                    "url": "https://i.ytimg.com/vi/b6Oe2puTdMQ/mqdefault.jpg",
                    "width": 320,
                  },
                  "standard": Object {
                    "height": 480,
                    "url": "https://i.ytimg.com/vi/b6Oe2puTdMQ/sddefault.jpg",
                    "width": 640,
                  },
                },
                "title": "React File Uploader With Express (Using React Hooks)",
              },
              "status": Object {
                "privacyStatus": "public",
              },
            },
          ]
        }
      />
      <div
        style={
          Object {
            "margin": "0 auto",
            "width": "50%",
          }
        }
      >
        <MoreButton
          onClickMore={[Function]}
        />
      </div>
    </div>
  </div>
</Fragment>
`;
